jobs/payment-jobs/tasks/cfs_create_invoice_task.py

```diff
@@ -49,6 +53,92 @@ def create_invoices(cls):
+            .filter(InvoiceModel.payment_method_code == PaymentMethod.INTERNAL.value) \
+            .filter(InvoiceModel.invoice_status_code == InvoiceStatus.APPROVED.value) \
+            .filter(InvoiceModel.routing_slip is not None) \
+            .order_by(InvoiceModel.created_on.asc()).all()
+
+        current_app.logger.info(f'Found {len(invoices)} to be created in CFS.')
+        receipt_response = {}
+        for invoice in invoices:
+            # Create a CFS invoice
+            current_app.logger.debug(f'Creating cfs invoice for invoice {invoice.id}')
+            routing_slip = RoutingSlipModel.find_by_number(invoice.routing_slip)
+            routing_slip_payment_account: PaymentAccountModel = PaymentAccountModel.find_by_id(
+                routing_slip.payment_account_id)
+
+            # apply invoice to the active CFS_ACCOUNT which will be the parent routing slip
+            active_cfs_account = CfsAccountModel.find_effective_by_account_id(
+                routing_slip_payment_account.id)
+
+            invoice_response = CFSService.create_account_invoice(transaction_number=invoice.id,
+                                                                 line_items=invoice.payment_line_items,
+                                                                 cfs_account=active_cfs_account)
+            invoice_number = invoice_response.json().get('invoice_number', None)
+            receipt_applied_amount = 0
+            # a routing slip can have multiple receipts
+            cfs_accounts: List[CfsAccountModel] = CfsAccountModel.find_by_account_id(
+                routing_slip_payment_account.id)
+            # an invoice has to be applied to multiple receipts ; apply till the balance is zero
+            for cfs_account in cfs_accounts:
+                try:
+                    # apply receipt now
+                    current_app.logger.debug(f'Apply receipt {cfs_account.id} on invoice {invoice_number} '
+                                             f'for routing slip {routing_slip.number}')
+                    receipt_response = CFSService.apply_receipt(cfs_account, routing_slip.number,
+                                                                invoice_number)
+
+                    # Create receipt.
+                    receipt = Receipt()
+                    receipt.receipt_number = receipt_response.json().get('receipt_number', None)
+                    # TODO verify if paybc response has a dollar
+                    receipt_amount = receipt_response.json().get('receipt_amount', None).replace('$', '')
+                    receipt.receipt_amount = receipt_amount
+                    receipt.invoice_id = invoice.id
+                    receipt.receipt_date = datetime.now()
+                    receipt.flush()
+
+                    # check total applied amount reaches the invoice amount , if so stop the apply
+                    receipt_applied_amount += receipt_amount
```

Comment:
This check has to change. Receipt applied amount will have the amount applied to other invoices too. So here do a GET on invoice in CFS and check if due amount is 0.